portfolio/src/main/webapp/typing.js

```diff
@@ -0,0 +1,58 @@
+// Type Writer JS Animation
+
+class TyperWriter {
+  constructor(elements, words, waitTime = 3000) {
+    this.elements = elements;
+    this.words = words;
+    this.wordIndex = 0;
+    this.txt = "";
+    this.waitTime = parseInt(waitTime, 10); //Base 10
+    this.typeFunc();
+    this.isDelete = false;
+  }
+}
+//typing main function
+TyperWriter.prototype.typeFunc = function () {
+  console.log(this.words.length);
+  const current_index = this.wordIndex % this.words.length;
+  const full_word = this.words[current_index];
+
+  if (this.isDelete) {
+    //Remove character
+    this.txt = full_word.substring(0, this.txt.length - 1);
+  } else {
+    //Add character
+    this.txt = full_word.substring(0, this.txt.length + 1);
+  }
+
+  //Output on website
+  this.elements.innerHTML = '<span class = "text">' + this.txt + "</span>";
```

Comment:
Instead of creating a span here, I would but the span inside the HTML and then set `span.textContent = this.txt`.

textContent is not treated as raw HTML, so it prevents XSS/HTML injection attacks. That's unlikely to happen here because txt is hardcoded, but better to be safe than sorry. (It's also slightly faster, but that doesn't matter as much.)

https://developer.mozilla.org/en-US/docs/Glossary/Cross-site_scripting